python/stack_plots.py

```diff
@@ -368,74 +465,158 @@ def draw_plot(sampleDict, plotname, logY=True, logX=False, plotData=False, doRat
+        h_axis_ratio.GetYaxis().SetLabelSize(0.12)
+        h_axis_ratio.GetYaxis().CenterTitle()
+        h_axis_ratio.GetYaxis().SetTickLength(0.02)
+        h_axis_ratio.GetXaxis().SetLabelSize(0)
+        h_axis_ratio.GetXaxis().SetTitle("")
+        h_axis_ratio.GetXaxis().SetTickSize(0.06)
+
+        line = ROOT.TLine(h_axis.GetXaxis().GetBinLowEdge(1), 1.0, h_axis.GetXaxis().GetBinUpEdge(h_axis.GetNbinsX()), 1.0)
+
+        pads.append(ROOT.TPad("1","1",0.0,0.18,1.0,1.0))
+        pads.append(ROOT.TPad("2","2",0.0,0.0,1.0,0.19))
+        pads[0].SetTopMargin(0.08)
+        pads[0].SetBottomMargin(0.13)
+        pads[0].SetRightMargin(0.05)
+        pads[0].SetLeftMargin(0.10)
+        pads[1].SetRightMargin(0.05)
+        pads[1].SetLeftMargin(0.10)
+        pads[0].Draw()
+        pads[1].Draw()
+        pads[1].cd()
+        pads[1].SetTickx()
         if logX:
-            if ratioplot.GetXaxis().GetBinLowEdge(1)<=0.0:
-                ratioplot.GetXaxis().SetRangeUser(1.0, ratioplot.GetXaxis().GetBinUpEdge(ratioplot.GetNbinsX()))
-            pad2.SetLogx()
-        ratioplot.Draw("E0")
+            h_axis_ratio.GetXaxis().SetMoreLogLabels()
+            pads[1].SetLogx()
+        h_axis_ratio.Draw("")
+        g_ratio_unc.Draw("SAME,2")
+        g_ratio.Draw("SAME,P0")
+        #
+        line.SetLineStyle(2)
+        line.SetLineColor(sampleLineColor["data"])
+        line.SetLineWidth(1)
+        line.Draw("SAME")
+        #
+        pads[1].RedrawAxis()
+
     else:
-        pad1 = ROOT.TPad("pad1","pad1",0,0,1,1)
-        pad1.Draw()
+        pads.append(ROOT.TPad("1","1",0,0,1,1))
+        pads[0].Draw()
 
-    pad1.cd()
+    pads[0].cd()
     if logY:
-        pad1.SetLogy()
+        pads[0].SetLogy()
     if logX:
-        pad1.SetLogx()
+        h_axis.GetXaxis().SetMoreLogLabels()
+        pads[0].SetLogx()
 
 
     #plot data, stack, signal, data  
-    stack.Draw("HIST")
-    if logX:
-        if stack.GetXaxis().GetBinLowEdge(1)<=0.0:
-            stack.GetXaxis().SetRangeUser(1.0, stack.GetXaxis().GetBinUpEdge(stack.GetXaxis().GetNbins()))
+    h_axis.GetYaxis().SetTitleSize(0.04)
+    h_axis.GetXaxis().SetTitleSize(0.04)
+    h_axis.GetXaxis().SetTitleOffset(1.25)
     if "cutflow" in plotname:
-        stack.GetXaxis().SetLabelSize(0.023)
+        h_axis.GetXaxis().SetLabelSize(0.023)
     else:
-        stack.GetXaxis().SetTitle(totalSM.GetXaxis().GetTitle())
-    stack.GetYaxis().SetTitle(totalSM.GetYaxis().GetTitle())
+        h_axis.GetXaxis().SetTitle(totalSM.GetXaxis().GetTitle())
     if args.shape:
-        stack.GetYaxis().SetTitle("A.U.")
-    stack.GetYaxis().SetLabelSize(0.03)
-    stack.GetYaxis().SetMaxDigits(3)
+        h_axis.GetYaxis().SetTitle("A.U.")
+    else:
+        h_axis.GetYaxis().SetTitle(totalSM.GetYaxis().GetTitle())
+    h_axis.GetYaxis().SetLabelSize(0.03)
+    h_axis.GetYaxis().SetMaxDigits(3)
+    h_axis.Draw("")
+    stack.Draw("HIST,SAME")
+    g_unc.Draw("SAME,2")
```

Comment:
What does the option `2` do?

Reply:
Option "2" allows to draw errors as rectangles (here rendered as dashes/grey area).